### PR TITLE
project-happy-thoughts-api Elle Frankenberg

### PR DESCRIPTION
**Pull request by ElleFrankenberg**

### CODE REVIEW

**Review by idasylven — APPROVED** [review]
Love how you made the effort to connect the backend with the previous frontend project! And another round of applause for going the extra length and applying a delete functionality to a thought that has been posted 👏🏻

We both had a very similar approach to this project, apart from some different naming and the fact that you customised your status messages in a bit more - very nice 👩‍🎤  ✉️  

Maybe you can add some explanatory comments if you prefer, and/or delete all the comments from the original repo (just one line is this case, nr 40) for an even cleaner server.js file.    

Another week of backend approved! 
Hooray! 🔙. 🔚. 🥂

File: server.js [hunk 1]
```diff
@@ -1,26 +1,104 @@
...
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const thoughtSchema = new mongoose.Schema({
+ message: {
+   type: String,
+   required: true,
+   trim: true,
+   minlength: 5,
+   maxlength: 140
```
Comment (KarolCislowski) [inline server.js:19]: nicely configured field 😻

File: server.js [hunk 2]
```diff
@@ -1,26 +1,104 @@
...
+    res.send(listEndpoints(app))
+  } catch (error) {
+    res.status(404).json({ error: 'Page not found' })
+  }
+})
+
+
+app.get('/thoughts', async (req, res) => {
+  try {
+    const recentThoughts = await Thought.find().sort({ createdAt: -1 }).limit(20)
```
Comment (KarolCislowski) [inline server.js:52]: 👍🏻

File: server.js [hunk 3]
```diff
@@ -1,26 +1,104 @@
...
+    res.status(400).json({ message: 'Could not save', error })
+  }
+})
+
+
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  const { thoughtId } = req.params
+
+  try {
+    const updatedLike = await Thought.findByIdAndUpdate(thoughtId, {$inc: { hearts: 1 } }, { new: true } )
```
Comment (KarolCislowski) [inline server.js:78]: 💯

File: server.js [hunk 4]
```diff
@@ -1,26 +1,104 @@
...
+    const deletedThought = await Thought.findByIdAndDelete(thoughtId)
+    if(deletedThought) {
+      res.json(deletedThought)
+    } else {
+      res.status(404).json({ error: 'No thought with matching id' })
+    }
+  } catch (error) {
+    res.status(400).json({ error: 'Could not delete the thought' })
+  }
 })
```
Comment (KarolCislowski) [inline server.js:102]: Nice but this endpoint need to check if you are the author of this message or any kind of admin. Always restrict access to the endpoints that are able to modify your data in DB

**Review by KarolCislowski — APPROVED** [review]
Great API 🦄

**Pull request closed without merging**